File: 8/main.c
#include <stdlib.h>
#include <stdio.h>

typedef struct fraction{
  long num;
  long denom;
} Fraction;

long gcd(long a, long b);


long gcd(long a, long b)
{
  long tmp = 0;
  if (b > a)
  {
    tmp = b;
    b = a;
    a = tmp;
  }  
  while(b != 0)
  {
    tmp = b;
    b = a % b;
    a = tmp;
  }
  return a;
}

Fraction simplify(Fraction a)
{
  if(a.num == 0 && a.denom == 0)
  {
    return a;
  }
  else
  {
    Fraction rtn;
    long GCD = gcd(a.num, a.denom);
    rtn.num = a.num / GCD;
    rtn.denom = a.denom / GCD;

    return rtn;
  }
}


Fraction makeFrac(long num, long denom)
{
  Fraction rtn;
  rtn.num = num;
  rtn.denom = denom;
  rtn = simplify(rtn);
  return rtn;
}

Fraction addFrac(Fraction a, Fraction b)
{
  Fraction rtn;
  rtn.num = a.num * b.denom + a.denom * b.num;
  rtn.denom = a.denom * b.denom;

  return simplify(rtn);
}


Fraction multFrac(Fraction a, Fraction b)
{
  Fraction rtn;
  rtn.num = a.num * b.num;
  rtn.denom = a.denom * b.denom;

  return simplify(rtn);
}



int main()
{
  int D = 0;
  int N = 0;
  Fraction out = makeFrac(0,0);
  
  while(scanf("%D %D", &D, &H))
  {
    for(; N <= pow(6, D), N++)
    
    

  }

  //printf("%ld", sizeof(long));
  Fraction f = makeFrac(36, 81);
  
  printf("%ld/%ld\n", f.num, f.denom);
}

File: 8/Main.java
import java.util.Scanner;
import java.math.BigInteger;


public class Main
{
  
  private class Fraction
  {
    BigInteger num;
    BigInteger denom;

    static final Fraction ZERO = new Fraction(0, 1);
    
    public Fraction(BigInteger a, BigInteger b)
    {
      if(a.equals(BigInteger.ZERO) && b.equals(BigInteger.ZERO))
      {
        num = 0;
        denom = 1;
      }
      else
      {
        BigInteger GCD = a.gcd(b);
        num = a.divide(GCD);
        denom = b.divide(GCD)
      }
    }
    
    public Fraction add(Fraction other)
    {
      BigInteger a = num;
      BigInteger b = denom;
      BigInteger c = other.num;
      BigInteger d = other.denom;

      BigInteger newNum = a.add(d).multiply(b.add(c));
      BigInteger newDenom = b.multiply(d);
      BigInteger GCD = newNum.gcd(newDenom);
      return new Fraction(newNum.divide(GCD), newDenom.divide(GCD));
    }

    public Fraction multiply(Fraction other)
    {
      BigInteger a = num;
      BigInteger b = denom;
      BigInteger c = other.num;
      BigInteger d = other.denom;

      BigInteger newNum = a.multiply(c);
      BigInteger newDenom = b.multiply(d);

      BigInteger GCD = newNum.gcd(newDenom);
      return new Fraction(newNum.divide(GCD), newDenom.divide(GCD));
    }

    public void print()
    {
      String printstr = "";
      if(num.equals(BigInteger.ZERO))
      {
        printstr = "0";
      }
      else if(num.equals(BigInteger.ONE) && denom.equals(BigInteger.ONE))
      {
        printstr = "1";
      }
      else
      {
        printstr = num.toString();
        printstr += "/";
        printstr += denom.toString();
      }
      System.out.println(printstr);
    }
  }

  public static void main(String args[])
  {
    Scanner in = new Scanner(System.in);
    in.useDelimiter("\n");

    while(in.hasNext())
    {
      String instr = in.next();
      String[] inNums = instr.split("\\s+");
      int D = inNums[0];
      int N = inNums[1];
       
      Fraction calcVal = Fraction.ZERO;
      for(int i = N; i <= 6*D; i++)
      {
        calcVal = calcVal.add(calc(D, i));
      }

      calcVal.print();
    }
  }
  
  public Fraction calc(int D, int N)
  {

    if(D == 1)
    {
      if(N >= 1 && N <= 6)
        return new Fraction(1, 6);
      else
        return Fraction.ZERO;
    }
    else
    {
      Fraction rtn = Fraction.ZERO;
      for(int i = 1; i < N; i++)
      {
        rtn = rtn.add(calc(1, i).multiply(calc(D-1, N-i)));
      }
      return rtn;
    }

    
  }
}

